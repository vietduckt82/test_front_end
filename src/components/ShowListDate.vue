<template>
    <div>
        <v-container>
            <v-row>
                <v-col cols="12" sm="12">
                <h3>All Day</h3>
                <v-data-table
                    :headers="headers"
                    :items="data"
                    :items-per-page="10"
                    :sort-by.sync="sortBy"
                    :sort-desc.sync="sortDesc"
                    id:list_page
                >
                    <template v-slot:[`item.Date`]="{ item }">
                    <span>{{ item.Date }}</span>
                    </template>
                </v-data-table>
                </v-col>
            </v-row>
        </v-container>
    </div>
</template>
<script>
export default {
    props: {
        data: {
        type: Array,
        },
    },
    data() {
        return {
            headers: [
                { text: "Country", value: "Country", sortable: true },
                { text: "Confirmed", value: "Confirmed", sortable: true },
                { text: "Deaths", value: "Deaths" },
                { text: "Recovered", value: "Recovered" },
                { text: "Active", value: "Active" },
                { text: "Date", value: "Date" },
            ],
            sortBy: "Date",
            sortDesc: true,
        };
    },
}
</script>