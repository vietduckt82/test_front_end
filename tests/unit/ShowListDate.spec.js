import { mount  } from "@vue/test-utils";
import ShowListDate from "@/components/ShowListDate.vue";

describe("ShowListDate.spec.js", () => {
    let cmp;
    beforeEach(() => {
		cmp = mount(ShowListDate, {
			// Beaware that props is overriden using `propsData`
        propsData: {
          data: [
            {
              Country: "Afghanistan",
              CountryCode: "AF",
              Date: "2020-10-31T05:00:38Z",
              NewConfirmed: 66,
              NewDeaths: 1,
              NewRecovered: 19,
              Slug: "afghanistan",
              TotalConfirmed: 41334,
              TotalDeaths: 1533,
              TotalRecovered: 34258
            },
          ],
        },
      });
    });
    
    it('has received ["Cat"] as the message property', () => {
      expect(cmp.vm.data).toEqual([
        {
          Country: "Afghanistan",
          CountryCode: "AF",
          Date: "2020-10-31T05:00:38Z",
          NewConfirmed: 66,
          NewDeaths: 1,
          NewRecovered: 19,
          Slug: "afghanistan",
          TotalConfirmed: 41334,
          TotalDeaths: 1533,
          TotalRecovered: 34258
        },
      ]);
    });
    it("has the expected html structure", () => {
      expect(cmp.element).toMatchSnapshot();
    });
    it("is a Home component", () => {
      expect(cmp.is(ShowListDate)).toBe(true);
    });
    // Vue instance
    it("Home is a vue instances", () => {
      expect(cmp.isVueInstance()).toBe(true);
    });
});