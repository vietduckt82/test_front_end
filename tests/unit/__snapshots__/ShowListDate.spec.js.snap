// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowListDate.spec.js has the expected html structure 1`] = `
<div>
  <v-container>
    <v-row>
      <v-col
        cols="12"
        sm="12"
      >
        <h3>
          All Day
        </h3>
         
        <v-data-table
          headers="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
          id:list_page=""
          items="[object Object]"
          items-per-page="10"
          sort-by="Date"
          sort-desc="true"
        />
      </v-col>
    </v-row>
  </v-container>
</div>
`;
